core/binding/src/store/primitive.rs

```diff
@@ -0,0 +1,205 @@
+use std::io::Cursor;
+use std::mem;
+
+use byteorder::{BigEndian, ReadBytesExt, WriteBytesExt};
+use bytes::Bytes;
+use cita_trie::DB as TrieDB;
+
+use protocol::traits::{ServiceState, StoreBool, StoreString, StoreUint64};
+use protocol::types::Hash;
+use protocol::ProtocolResult;
+
+use crate::state::GeneralServiceState;
+use crate::store::StoreType;
+use crate::BindingError;
+
+pub struct DefaultStoreBool<DB: TrieDB> {
+    state: GeneralServiceState<DB>,
+    key:   Hash,
+}
+
+impl<DB: TrieDB> DefaultStoreBool<DB> {
+    pub fn new(state: GeneralServiceState<DB>, var_name: &str) -> Self {
```

Comment:
Should not pass a concrete `ServiceState` implementation.

In addition, `GeneralServiceState` does not currently support `clone`. How does it share state between modules?

Reply:
Changed to `pub struct DefaultStoreBool<S: ServiceState>` and use `state : Rc<RefCell<S>>`